src/k8s-engine/kim.ts

```diff
@@ -132,6 +132,9 @@ class Kim extends EventEmitter {
   }
 
   async runKimCommand(args: string[], sendNotifications = true): Promise<childResultType> {
+    if (!args.includes('--context') && !args.includes('x')) {
```

Comment:
Argument parsing is tricky:
- `runKimCommand(['--context=foo'])` should match.
- `runKimCommand(['-x', 'foo'])` should match.
- `runKimCommand(['build', '-t', 'x', '.'])` should not match.

Alternatively:
- Make this `private`
- Assert that we don't already have a context set instead (since nothing in this class does)
- And add it.

Reply:
Actually, because we completely control the structure* of `args` in all calls to `runKimCommand`, we can simply splice in `['--context', 'rancher-desktop']` near the start of the args list. And I'd rather not use `splice`, as it can modify the caller's `args` parameter unexpectedly.

[*] - we don't control tag names, image names, etc. to in the `args` list, just their positions, hence the statement that we do control the structure.

Reply:
Also that should have been `-x`, not `x`.